core/templates/ordered_hash_map.h

```diff
@@ -204,12 +293,11 @@ class OrderedHashMap {
 	Element insert(const K &p_key, const V &p_value) {
 		typename InternalList::Element **list_element = map.getptr(p_key);
 		if (list_element) {
-			(*list_element)->get().second = p_value;
+			(*list_element)->get().value = p_value;
 			return Element(*list_element);
 		}
-		typename InternalList::Element *new_element = list.push_back(Pair<const K *, V>(nullptr, p_value));
-		typename InternalMap::Element *e = map.set(p_key, new_element);
-		new_element->get().first = &e->key();
```

Comment:
Again, i think this is done this way (previous line 210 and 212), is to avoid a double key copy.    In your version there are two copies of p_key, as the list's key is no longer a pointer.

Reply:
That makes sense, however, since the `key` member is read-only it won't let me assign it later.

Reply:
Nevermind, I can make it work. However the for loop won't be consistent with `Map`.

```c++
for (const KeyValue<const String *, List<Ref<InputEvent>>> &E : builtins) {
	...

	String action_name = "input/" + *(E.key);

	...
}
```

As opposed to:

```c++
for (const KeyValue<String, List<Ref<InputEvent>>> &E : builtins) {
	...

	String action_name = "input/" + E.key;

	...
}
```

Is that acceptable?

Reply:
I don't think so, we would need to keep range iteration consistent across containers.  Maybe the list maintains a reference of the key stored in the hash map?  I will try and check this out this afternoon and help.